protostar/starknet/cheatable_syscall_handler.py

```diff
@@ -202,36 +179,28 @@ def _deploy(
             constructor_calldata=constructor_calldata,
             deployer_address=deployer_address,
         )
-        # BEGIN: PROTOSTAR_MODIFICATION — update mappings
+
+        # --- MODIFICATIONS START --- # TODO
         self.cheatable_state.contract_address_to_class_hash_map[
             contract_address
         ] = class_hash
-        # END: PROTOSTAR_MODIFICATION
+        # --- MODIFICATIONS END ---
 
         # Initialize the contract.
         class_hash_bytes = to_bytes(class_hash)
-        future = asyncio.run_coroutine_threadsafe(
-            coro=initialize_contract_state(
-                state=self.state,
-                class_hash=class_hash_bytes,
-                contract_address=contract_address,
-            ),
-            loop=self.loop,
+        self.sync_state.deploy_contract(
+            contract_address=contract_address, class_hash=class_hash_bytes
         )
-        future.result()
 
         self.execute_constructor_entry_point(
             contract_address=contract_address,
             class_hash_bytes=class_hash_bytes,
             constructor_calldata=constructor_calldata,
         )
 
-        # Update deployed contract addresses.
-        self.deployed_contracts.append(contract_address)
-
         return contract_address
 
 
 class CheatableHintsWhitelist(HintsWhitelist):
-    def verify_hint_secure(self, hint, reference_manager):  # type: ignore
-        return True
+    def verify_hint_secure(self, hint, reference_manager):
+        pass
```

Comment:
Did compiler stop complaining about Protostar's cheatcodes?

Reply:
Nope, that is caused by the fact that cheatcodes are not injected into VM locals and this is separate issue.

Reply:
But after 4d1eb4f1fb5036de4bdacb3e5936d0ad59874b2f this is fixed. Done.